### PR TITLE
Expand skip spaces

### PR DESCRIPTION
**Pull request by fernape**
Expand `skipSpaces` so it handles new lines and tabs. This will be useful since queries in code tend to be indented for clarity. Since it now handles more than spaces, rename the function to `skipBlanks`

### CODE REVIEW

File: internal/parse/parser_internal_test.go [hunk 1]
```diff
@@ -29,11 +29,12 @@ func TestRunTable(t *testing.T) {
...
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "abc    d", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "     abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "  \t  abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "\t  abcd", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{false}, input: "", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{false}, input: "abc    d", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{true}, input: "     abcd", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{true}, input: "  \t  abcd", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{true}, input: "\t  abcd", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{true}, input: "\n\t  abcd", data: []string{}},
```
Comment (TheSignPainter98) [inline internal/parse/parser_internal_test.go]: There's no test for `\r` here :)

File: internal/parse/parser.go [hunk 2]
```diff
@@ -177,15 +177,18 @@ func (p *Parser) skipByteFind(b byte) bool {
...
 }
 
-// skipSpaces advances the parser jumping over consecutive spaces. It stops when
-// finding a non-space character. Returns true if the parser position was
-// actually changed, false otherwise.
-func (p *Parser) skipSpaces() bool {
+// skipBlanks advances the parser jumping over consecutive spaces, tabs or new
+// lines. It stops when finding a non-blank character. Returns true if the
+// parser position was actually changed, false otherwise.
+func (p *Parser) skipBlanks() bool {
```
Comment (TheSignPainter98) [inline internal/parse/parser.go]: The middle sentence seems redundant to me. Perhaps something like this?
```go
// skipBlanks advances the parser past spaces, tabs and newlines. Returns whether
// the parser position was changed.
```

File: internal/parse/parser_internal_test.go (hunk 1 above)
Comment (fernape) [reply internal/parse/parser_internal_test.go]: Done.

Thanks!

File: internal/parse/parser.go (hunk 2 above)
Comment (fernape) [reply internal/parse/parser.go]: Simpler and clearer. Thanks!

File: internal/parse/parser_internal_test.go [hunk 3]
```diff
@@ -29,11 +29,16 @@ func TestRunTable(t *testing.T) {
 		{bytef: p.skipByteFind, result: []bool{false, true, true}, input: "abcde", data: []string{"x", "b", "c"}},
 		{bytef: p.skipByteFind, result: []bool{true, false}, input: "abcde ", data: []string{" ", " "}},
 
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "abc    d", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "     abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "  \t  abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "\t  abcd", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{false}, input: "", data: []string{}},
+		{stringf0: p.skipBlanks, result: []bool{false}, input: "abc    d", data: []string{}},
```
Comment (solmyrcan) [inline internal/parse/parser_internal_test.go]: Would it be relevant to have test cases for `\n \r \t` also mixed up in the whitespaces as in here:
`input: "abc  \t  d"`
Just used `\t` here, but maybe others as well?

Comment (fernape) [reply internal/parse/parser_internal_test.go]: Those should be covered with the tests around line 37 and such.

File: internal/parse/parser.go [hunk 4]
```diff
@@ -177,15 +177,17 @@ func (p *Parser) skipByteFind(b byte) bool {
...
-		if p.input[p.pos] != ' ' {
-			break
-		}
+	blanks := map[byte]bool{
+		' ':  true,
+		'\t': true,
+		'\r': true,
+		'\n': true,
+	}
+	for p.pos < len(p.input) && blanks[p.input[p.pos]] {
```
Comment (niemeyer) [inline internal/parse/parser.go]: This is a longer and much more expensive version of this:
```
for ... {
    switch p.input[p.pos] {
    case ' ', '\t', ...:
        ...
    }
}
```

**Review by niemeyer — CHANGES_REQUESTED** [review]
Idea is good, but the implementation could be tuned a bit per notes below.

File: internal/parse/parser.go (hunk 4 above)
Comment (fernape) [reply internal/parse/parser.go]: Done. Thanks.

**Pull request closed without merging**